lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -642,6 +680,330 @@ def get_fq_stats(self, reads_object, file_path):
+        reads_source: one of 'shock', 'web' or 'staging'
+        download_type: one of ['Direct Download', 'FTP', 'DropBox', 'Google Drive']
+        user_id: current token user
+        """
+
+        # return values
+        fwdpath = None
+        revpath = None
+        fwdname = None
+        revname = None
+  
+        if reads_source == 'shock':
+            # Grab files from Shock
+            fwdid = fwd
+            revid = rev
+            dfu = DataFileUtil(self.callback_url)
+            fileinput = [{'shock_id': fwdid,
+                          'file_path': self.scratch + '/fwd/',
+                          'unpack': 'uncompress'}]
+            if revid:
+                fileinput.append({'shock_id': revid,
+                                  'file_path': self.scratch + '/rev/',
+                                  'unpack': 'uncompress'})
+            self.log('downloading reads file(s) from Shock')
+            files = dfu.shock_to_file_mass(fileinput)
+            fwdpath = files[0]["file_path"]
+            fwdname = files[0]["node_file_name"]
+            if revid:
+                revpath = files[1]["file_path"]
+                revname = files[1]["node_file_name"]
+        elif reads_source == 'web':
+            # TODO: Tian move _download_web_file to DFU
+            fwdpath = self._download_web_file(fwd, download_type)
+            revpath = self._download_web_file(rev, download_type) if rev else None
+        elif reads_source == 'staging':
+            # TODO: Tian move _download_staging_file to DFU
+            fwdpath = self._download_staging_file(user_id, fwd)
+            revpath = self._download_staging_file(user_id, rev) if rev else None
+        elif reads_source == 'local':
+            pass
+        else:
+            raise ValueError("Unexpected reads_source value. reads_source: %s" % reads_source)
+
+        returnVal = {'fwdpath': fwdpath, 
+                     'revpath': revpath,
+                     'fwdname': fwdname, 
+                     'revname': revname}
+
+        return returnVal
+
+    def _generate_validation_error_message(self, reads_source, actualpath, fwdpath, revpath, fwdname, revname, fwdid, revid, fwdurl, revurl, fwdstaging, revstaging):
+        validation_error_message = "Invalid FASTQ file - Path: " + actualpath + "."
+        if reads_source == 'shock':
+            if revid:
+                validation_error_message += (
+                    " Input Shock IDs - FWD Shock ID : " +
+                    fwdid + ", REV Shock ID : " + revid +
+                    ". FWD File Name : " + fwdname +
+                    ". REV File Name : " + revname +
+                    ". FWD Path : " + fwdpath +
+                    ". REV Path : " + revpath + ".")
+            else:
+                validation_error_message += (" Input Shock ID : " + 
+                                        fwdid + ". File Name : " + fwdname + ".")
```

Comment:
I did
"validation_error_message = "Invalid FASTQ file - Path: " + actualpath + ".""

at the beginning.